languages/javascript/reference/keywords/import.md

```diff
@@ -1 +1,24 @@
 # `import`
+
+The `import` statement is used to load [functions][concept-function], [variables][concept-variable] and [classes][concept-class] from other [modules][concept-modules] into the current file.
+
+It comes in multiple versions that can load the default [`export`][concept-export], some or all named exports and rename them on the fly:
+
+```js
+import defaultExport from "module-name"; // Load one default export
+import * as name from "module-name"; // Load all named exports for access via `name.namedExport`
+import { myExport } from "module-name"; // Only imports the myExport named export.
+import { myExport as expert } from "module-name"; // Only imports the myExport but aliases it to expert
+```
+
+The `module-name` in the example can be either a [NPM][npm]-installed package – those typically reside in the `node_modules/` directory – or a relative link, such as `./myFile.js`.
```

Comment:
It can also be a url. What you can use here is actually determined by the module resolver, which is different in different contexts (read: node, browsers, deno)

Reply:
I am always learning new stuff. Never heard of deno before, but sounds interesting :)

Reply:
Maybe we should also mention, that `import` is used in babel context, and in Node.js it the `required` function that does it there…

Reply:
First things first. `import` and `require` are _both_ ECMAScript constructs. Node now supports _both_. The `import` keyword is different from the `import()` function. Both are a thing in ECMAScript. Node supports both. 

Now here comes the fun part:

- import in CommonJS: `require`
- export in CommonJS: `module.exports`
- import in ES6: `import`
- export in ES6: `export`

CommonJS is not a part of ECMAScript and was designed to have modules _outside_ of the web context. Bundlers, such as rollup, webpack etc. inject this function so that you can actually use these in the browser context (read: so that `require` actually resolves a module, either in-line or lazy via an xhr/fetch).

Finally, the _dynamic `import()` _ call was something supported by webpack (Remember `System.import`?) but is actually part of the standard. It allows you to _dynamically_ load a module, and it **will work** in scripts (read: no `type="module"`!!). It's defined here: https://tc39.es/ecma262/#sec-import-calls.

------

I think we need a section that tells about module systems that are non-standard (requirejs, amd, commonjs, umd) that all pre-date the standard (es6 modules) and that `require` is indeed an artifact of those systems. Many tools such as `babel` and `tsc` will correctly interpret these global function calls.

Reply:
Thanks for that.
I think that topic can and should be explained in more depths in the `info/module.md`. So I added some words and examples here and a link to that page for further reading.